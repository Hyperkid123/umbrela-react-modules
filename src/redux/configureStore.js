import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
