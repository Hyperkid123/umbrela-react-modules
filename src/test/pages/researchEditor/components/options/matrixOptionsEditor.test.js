import React from 'react'
import MatrixOptionsEditor, {MatrixOptionsEditor as Snapshot} from '../../../../../pages/researchEditorModule/components/options/matrixOptionsEditor';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('Matrix options editor component', () => {
  const initialState  = {
    questions: {
      activeQuestion: {
        questionId: 1
      }
    },
    options: {
      options: []
    },
    ui: {
      draggingElement: false
    },
    locale: {
      languages: [{
        name: "English",
        code: 'en',
        active: true,
      }],
      translations: {}
    }
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<MatrixOptionsEditor store={store}/>);
    expect(wrapper).toBeDefined();
  })
  it('should render correclty', () => {
      const tree = shallow(
        <Snapshot
          getOptions={jest.fn()}
          activeQuestion={{questionId: 1}}
          options={[
            {optionOrder: 1, optionId: 1, optionType: 'ColumnOption'},
            {optionOrder: 10, optionId: 10, optionType: 'RowOption'}
          ]}
          translate={jest.fn()}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  it('should call getOptions correctly', () => {
    const getOptions = jest.fn();
    const wrapper = shallow(
      <Snapshot
        getOptions={getOptions}
        activeQuestion={{questionId: 1}}
        options={[
          {optionOrder: 1, optionId: 1, optionType: 'ColumnOption'},
          {optionOrder: 10, optionId: 10, optionType: 'RowOption'}
        ]}
        translate={jest.fn()}
      />
    );
    expect(getOptions.mock.calls.length).toEqual(1);
    wrapper.instance().componentWillReceiveProps({activeQuestion: {questionId: 2}});
    expect(getOptions.mock.calls.length).toEqual(2);
  })
});
