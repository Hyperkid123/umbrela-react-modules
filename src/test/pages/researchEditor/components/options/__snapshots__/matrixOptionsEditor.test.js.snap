// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matrix options editor component should render correclty 1`] = `
<div
  className="sc-bdVaJa fizmbb"
>
  <styled.div
    column={true}
  >
    <styled.div>
      <styled.h2 />
      <styled.ul>
        <Connect(OptionEditorItem)
          deleteOption={[Function]}
          index={1}
          key="1"
          option={
            Object {
              "optionId": 1,
              "optionOrder": 1,
              "optionType": "ColumnOption",
            }
          }
          optionOrder={1}
        />
      </styled.ul>
      <Connect(NewOption)
        optionType="ColumnOption"
      />
    </styled.div>
  </styled.div>
  <styled.div
    column={true}
  >
    <styled.div>
      <styled.h2 />
      <styled.ul>
        <Connect(OptionEditorItem)
          deleteOption={[Function]}
          index={10}
          key="10"
          option={
            Object {
              "optionId": 10,
              "optionOrder": 10,
              "optionType": "RowOption",
            }
          }
          optionOrder={10}
        />
      </styled.ul>
      <Connect(NewOption)
        optionType="RowOption"
      />
    </styled.div>
  </styled.div>
</div>
`;
